openapi/components/schemas/Quote/Quote.yaml

```
@@ -0,0 +1,29 @@
+type: object
+allOf:
+  - $ref: ./CommonQuote.yaml
+  - properties:
+      couponIds:
+        type: array
+        nullable: true
+        description: |-
+          A list of coupons to redeem on the customer and restrict to the accepted order.
+
+          Read more about [coupons
+          here](https://docs.rebilly.com/docs/dashboard/marketing/coupons-and-discounts/).
+
+          This parameter respects the following logic:
+
+          - When not passed then applied coupons will not be changed.
+
+          - When empty array passed then all applied coupon redemptions will be
+          canceled.
+
+          - When list of coupons is passed then not applied yet coupons will be
+          applied, already applied coupons will not change their state, applied
+          coupons that are not presented in passed list will be canceled.
+
+          If coupons will change before the quote is accepted the final discount
+          might differ.
```

issue: this description is out of date
```suggestion
          List of coupons to redeem on the customer and restrict to this order.
          
          For more information, see [Coupons](https://www.rebilly.com/docs/settings/coupons-and-discounts/).
          
          This parameter uses the following logic:
          - If this parameter is not supplied, applied coupons are not changed.
          - If an empty array is supplied, all applied coupon redemptions are canceled.
          - If a list of coupons is supplied, unapplied coupons in the list are applied.
          Coupons that have already been applied do not change state.
          Applied coupons that are not supplied in list are canceled.
          
          If the list of applied coupons on a pending order is changed by this parameter during an order update, the invoice for the order is reissued.
```